spec/models/state_file_ny_intake_spec.rb

```diff
@@ -0,0 +1,126 @@
+#  ny_mailing_apartment             :string
+#  ny_mailing_city                  :string
+#  ny_mailing_street                :string
+#  ny_mailing_zip                   :string
+#  ny_other_additions               :integer
+#  nyc_full_year_resident           :integer          default("unfilled"), not null
+#  occupied_residence               :integer          default("unfilled"), not null
+#  permanent_apartment              :string
+#  permanent_city                   :string
+#  permanent_street                 :string
+#  permanent_zip                    :string
+#  phone_number                     :string
+#  primary_birth_date               :date
+#  primary_email                    :string
+#  primary_first_name               :string
+#  primary_last_name                :string
+#  primary_middle_initial           :string
+#  primary_signature                :string
+#  property_over_limit              :integer          default("unfilled"), not null
+#  public_housing                   :integer          default("unfilled"), not null
+#  raw_direct_file_data             :text
+#  referrer                         :string
+#  refund_choice                    :integer          default("unfilled"), not null
+#  residence_county                 :string
+#  routing_number                   :string
+#  sales_use_tax                    :integer
+#  sales_use_tax_calculation_method :integer          default("unfilled"), not null
+#  school_district                  :string
+#  school_district_number           :integer
+#  source                           :string
+#  spouse_birth_date                :date
+#  spouse_first_name                :string
+#  spouse_last_name                 :string
+#  spouse_middle_initial            :string
+#  spouse_signature                 :string
+#  untaxed_out_of_state_purchases   :integer          default("unfilled"), not null
+#  created_at                       :datetime         not null
+#  updated_at                       :datetime         not null
+#  visitor_id                       :string
+#
+require "rails_helper"
+
+describe StateFileNyIntake do
+  describe "before_save" do
+    let(:intake) { create :state_file_ny_intake, untaxed_out_of_state_purchases: "yes", sales_use_tax_calculation_method: "manual", sales_use_tax: "350", household_fed_agi: 102_000 }
+
+    context "when untaxed_out_of_state_purchases changes to no" do
+      it "clears sales use tax calculation method and sales use tax" do
+        expect {
+          intake.update(untaxed_out_of_state_purchases: "no")
+        }.to change(intake, :sales_use_tax_calculation_method).to("unfilled")
+         .and change(intake, :sales_use_tax).to(nil)
+      end
+    end
+
+    context "when sales_use_tax_calculation_method changes to automated" do
+      it "calculates sales use tax" do
+        expect {
+          intake.update(sales_use_tax_calculation_method: "automated")
+        }.to change(intake, :sales_use_tax).to(29)
+      end
+    end
```

Comment:
This is great! I love that you put this on the model in a `before_save` hook. I think that makes a ton of sense and makes me feel confident that the data will always maintain integrity.